workflow/rules/bwa_mem.smk

```diff
@@ -0,0 +1,51 @@
+# vim: syntax=python tabstop=4 expandtab
+# coding: utf-8
+
+__author__ = "Jonas Almlöf"
+__copyright__ = "Copyright 2021, Jonas Almlöf"
+__email__ = "jonas.almlof@scilifelab.uu.se"
+__license__ = "GPL-3"
+
+
+rule bwa_mem:
+    input:
+        reads=["pre-alignment/merge/{sample}_{unit}_R1.fq.gz", "Pre-alignment/merge/{sample}_{unit}_R2.fq.gz"],
+        index=config["references"]["genome"],
+    output:
+        bam="alignment/bwa_mem/{sample}_{unit}.bam",
+    params:
+        index=config["references"]["genome"],
+        extra=config["bwa_mem"]["extra"],
+        sort=config["bwa_mem"].get("sort", "samtools"),
```

Comment:
I was wondering how `.get` works, does it use the default if the field is `""` or if it does not exist?

Reply:
If first argument does not exist in the hash it returns the second argument.

Reply:
hm, so empty string would be a valuable argument for get?

Reply:
A hash in python can't have  "" as a key so only the default value can be ""

Reply:
ok 👍